### PR TITLE
add jtdx

### PR DESCRIPTION
**Pull request by lu9dce**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.tech.JTDX.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>org.tech.JTDX</id>
+    <name>JTDX</name>
+    <provides>
+        <id>jtdx.desktop</id>
+    </provides>
+    <summary>jtdx (forked WSJT-x for amaterur radio communication)</summary>
```
Comment (hfiguiere) [inline org.tech.JTDX.metainfo.xml:8]:
```suggestion
    <summary>jtdx (forked WSJT-x for amateur radio communication)</summary>
```

**Comment by hfiguiere** [comment]
the shared-module git submodule is missing.

**Comment by lu9dce** [comment]
> the shared-module git submodule is missing.

ok no problem

modify it later
 tks!

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

File: org.tech.JTDX.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        url: https://github.com/Hamlib/Hamlib/releases/download/4.5.4/hamlib-4.5.4.tar.gz
+        sha256: b1aea97d6093990b77d5dc4bde6b9ca06183ddb7e24da7e2367a2acc957b7ac2
+
+  - name: jtdx
+    sources:
+      - type: archive
+        url: https://github.com/jtdx-project/jtdx/archive/159/jtdx-159.tar.gz
+      - type: file
+        path: org.tech.JTDX.metainfo.xml
+    buildsystem: cmake
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:66]:
```suggestion
    buildsystem: cmake-ninja
```

File: org.tech.JTDX.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  - name: hamlib
+    sources:
+      - type: archive
+        url: https://github.com/Hamlib/Hamlib/releases/download/4.5.4/hamlib-4.5.4.tar.gz
+        sha256: b1aea97d6093990b77d5dc4bde6b9ca06183ddb7e24da7e2367a2acc957b7ac2
+
+  - name: jtdx
+    sources:
+      - type: archive
+        url: https://github.com/jtdx-project/jtdx/archive/159/jtdx-159.tar.gz
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:63]: a sha256 checksum is required here.

File: org.tech.JTDX.yml (hunk 2 above)
Comment (lu9dce) [reply org.tech.JTDX.yml:66]: ok

File: org.tech.JTDX.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  - /include
+  - /share/man
+  - /share/doc
+  - /doc
+  - '*.a'
+  - '*.la'
+
+modules:
+  - shared-modules/libusb/libusb.json
+  - shared-modules/linux-audio/fftw3f.json
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:27]: you need to add the git submodule.

It's documented there https://github.com/flathub/shared-modules/

**Comment by hfiguiere** [comment]
bot, build org.tech.JTDX

File: .gitmodules
Comment (hfiguiere) [inline .gitmodules:1]: You didn't actually add the submodule. So it's still not working. You forgot to commit `shared-modules`

Comment (lu9dce) [reply .gitmodules:1]: i am comparing with wsjtx from flatpak

File: org.tech.JTDX.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      - type: archive
+        url: https://github.com/Hamlib/Hamlib/releases/download/4.5.4/hamlib-4.5.4.tar.gz
+        sha256: b1aea97d6093990b77d5dc4bde6b9ca06183ddb7e24da7e2367a2acc957b7ac2
+
+  - name: jtdx
+    sources:
+      - type: archive
+        url: https://github.com/jtdx-project/jtdx/archive/159/jtdx-159.tar.gz
+        sha256: 85b02fbe7f1d53e29dd592fc78e32a458a2facf6ae08bbd24ec2b1fe16155dcc
+      - type: file
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:56]: you removed the `path` for the file which is now missing.

Comment (lu9dce) [reply org.tech.JTDX.yml:56]: yes, I saw it!

hey thanks for the help!

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

**Comment by lu9dce** [comment]
wow 99%!!! build!

File: org.tech.JTDX.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
+app-id: org.tech.JTDX
+runtime: org.kde.Platform
+runtime-version: 5.15-22.08
+sdk: org.kde.Sdk
+command: jtdx
+rename-icon: jtdx
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:6]: it doesn't find an icon with this name.

Comment (lu9dce) [reply org.tech.JTDX.yml:6]: ok ! yes .. solving...

One question... I'm going to collaborate with more than 50 amateur radio programs

%99 uses hamlib .. my question is, wouldn't it be better to add a hamlib to the shared modules

to avoid having to indicate in so many programs to compile that library!

**Comment by hfiguiere** [comment]
it's installed in the wrong place:
```
-- Installing: /app/share/pixmaps/jtdx_icon.png
```

Also the desktop file needs to be renamed:
```
-- Installing: /app/share/applications/jtdx.desktop
```

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

File: org.tech.JTDX.yml (hunk 6 above)
Comment (hfiguiere) [reply org.tech.JTDX.yml:6]: this icon name doesn't match what's installed.

File: org.tech.JTDX.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      - type: file
+        path: org.tech.JTDX.metainfo.xml
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DWSJT_SKIP_MANPAGES=ON
+      - -DWSJT_GENERATE_DOCS=OFF
+    post-install:
+      - install -D org.tech.JTDX.metainfo.xml /app/share/metainfo/org.tech.JTDX.metainfo.xml
+      - mkdir -p /app/share/icons/hicolor/64x64/apps/
+      - cp /app/share/pixmaps/jtdx_icon.png /app/share/icons/hicolor/64x64/apps/jtdx_icon.png
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:65]:
```suggestion
      - install -Dm644 /app/share/pixmaps/jtdx_icon.png /app/share/icons/hicolor/64x64/apps/jtdx_icon.png
```

And we require at least 128x128 icon so this might not be enough.

Comment (lu9dce) [reply org.tech.JTDX.yml]: ok tks

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

File: org.tech.JTDX.yml [hunk 8]
```diff
@@ -0,0 +1,64 @@
+app-id: org.tech.JTDX
+runtime: org.kde.Platform
+runtime-version: 5.15-22.08
+sdk: org.kde.Sdk
+command: jtdx
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:5]:
```suggestion
command: jtdx
rename-icon: jtdx-icon
rename-desktop-file: jtdx.desktop
```

File: org.tech.JTDX.yml [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        sha256: 85b02fbe7f1d53e29dd592fc78e32a458a2facf6ae08bbd24ec2b1fe16155dcc
+      - type: file
+        path: org.tech.JTDX.metainfo.xml
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DWSJT_SKIP_MANPAGES=ON
+      - -DWSJT_GENERATE_DOCS=OFF
+    post-install:
+      - install -D org.tech.JTDX.metainfo.xml /app/share/metainfo/org.tech.JTDX.metainfo.xml
+      - install -Dm644 /app/share/pixmaps/jtdx_icon.png /app/share/icons/hicolor/64x64/apps/jtdx_icon.png
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:65]: I believe it's a 128x128 icon (this one is https://github.com/jtdx-project/jtdx/blob/master/icons/Unix/jtdx_icon.png) . Which

1. make this install path incorrect
2. would make the build succeed.

Comment (lu9dce) [reply org.tech.JTDX.yml]: ok 128x128!

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

File: org.tech.JTDX.yml [hunk 10]
```diff
@@ -0,0 +1,66 @@
+app-id: org.tech.JTDX
+runtime: org.kde.Platform
+runtime-version: 5.15-22.08
+sdk: org.kde.Sdk
+command: jtdx
+rename-icon: jtdx-icon
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:6]:
```suggestion
rename-icon: jtdx_icon
```

it was a typo.

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

**Comment by lu9dce** [comment]
If this program is going well.. I will already have it as a reference since I have a base of a soft that I know well how it is compiled and I will have the comparison against flatpak we will see if in the other I can use less commits .. The problem I have is my PC too old to compile locally.. I appreciate the help

**Comment by lu9dce** [comment]
Test !!!
Screensho

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

File: org.tech.JTDX.yml [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+      - ./b2 -j $FLATPAK_BUILDER_N_JOBS install
+
+  - name: hamlib
+    sources:
+      - type: archive
+        url: https://github.com/Hamlib/Hamlib/releases/download/4.5.4/hamlib-4.5.4.tar.gz
+        sha256: b1aea97d6093990b77d5dc4bde6b9ca06183ddb7e24da7e2367a2acc957b7ac2
+
+  - name: jtdx
+    sources:
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:53]: sources is usually last in the module.

File: org.tech.JTDX.yml [hunk 12]
```diff
@@ -0,0 +1,66 @@
+app-id: org.tech.JTDX
```
Comment (hfiguiere) [inline org.tech.JTDX.yml:1]: `tech.org` seems to be unrelated to this package.

this would make the app-id invalid.

Comment (lu9dce) [reply org.tech.JTDX.yml:1]: in readme https://github.com/jtdx-project/jtdx#readme

Project Web site:

http://www.jtdx.tech/

Project mailing  list:

https://jtdx.groups.io/g/main

app-id : tech.jtdx.JTDX ?

**Comment by lu9dce** [comment]
> `jtdx.tech` seem to now be a "parked" domain. you could ise `io.github.jtdx_project.JTDX` 

ok !!

**Comment by lu9dce** [comment]
> `jtdx.tech` seem to now be a "parked" domain. you could ise `io.github.jtdx_project.JTDX` 

ok !!  under review

**Comment by lu9dce** [comment]
bot, build org.tech.JTDX

**Comment by lu9dce** [comment]
bot, build io.github.jtdx_project.JTDX

File: COPYING
Comment (hfiguiere) [inline COPYING:1]: this is not necessary

File: README
Comment (hfiguiere) [inline README:1]: this doesn't provide any useful information.

**Comment by lu9dce** [comment]
bot, build io.github.jtdx_project.JTDX

File: io.github.jtdx_project.JTDX.yml [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+  - /man
+  - /include
+  - /share/man
+  - /share/doc
+  - /doc
+  - "*.a"
+  - "*.la"
+
+modules:
+  - shared-modules/libusb/libusb.json
```
Comment (hfiguiere) [inline io.github.jtdx_project.JTDX.yml:27]: you removed the .gitmodules which break the build.

File: io.github.jtdx_project.JTDX.metainfo.xml [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    <id>io.github.jtdx_project.JTDX</id>
+    <name>JTDX</name>
+    <provides>
+        <id>jtdx.desktop</id>
+    </provides>
+    <summary>jtdx (forked WSJT-x for amateur radio communication)</summary>
+    <developer_name>Arvo Järve</developer_name>
+    <description>
+        <p>
+            <![CDATA[JTDX is a forked from WSJT-X, a computer program dedicated to]]>
```
Comment (hfiguiere) [inline io.github.jtdx_project.JTDX.metainfo.xml:12]: this is incorrect. It should just be text.
